editors/code/src/main.ts

```diff
@@ -34,7 +35,13 @@ export async function activate(context: vscode.ExtensionContext) {
 
     const config = new Config(context);
 
-    const serverPath = await ensureServerBinary(config.serverSource);
+    vscode.workspace.onDidChangeConfiguration(() => ensureProperExtensionVersion(config).catch(log.error));
+
+    // Don't await the user response here, otherwise we will block the lsp server bootstrap
+    void ensureProperExtensionVersion(config).catch(log.error);
+
+    const serverPath = await ensureServerBinary(config);
```

Comment:
I am not entirely sure this is the right decision, since the user may end up looking at both notifications simultaneously:
![image](https://user-images.githubusercontent.com/36276403/76250202-83b67c80-624d-11ea-9afa-1881a4989571.png)

Reply:
I think that's [good enough](https://b1.culture.ru/c/793937.jpg)